packages/shadcn/src/index.tsx

```diff
@@ -24,41 +36,263 @@ import {
-  Toolbar,
-  ToolbarButton,
-  ToolbarSelect,
-  Menu,
-  MenuTrigger,
-  MenuDropdown,
-  MenuLabel,
-  MenuDivider,
-  MenuItem,
-  Panel,
-  PanelButton,
-  PanelFileInput,
-  PanelTab,
-  PanelTextInput,
-  Popover: Popover,
-  PopoverContent: PopoverContent,
-  PopoverTrigger: PopoverTrigger,
-  TextInput: TextInput,
-  Form: Form,
-  //   SuggestionMenuLoader: () => (
-  //     <Loader className={"bn-slash-menu-loader"} type="dots" />
-  //   ),
+export const components: Components = createComponentsContext({
+  FormattingToolbar: {
+    Root: Toolbar,
+    Button: ToolbarButton,
+    Select: ToolbarSelect,
+  },
+  ImagePanel: {
+    Root: Panel,
+    Button: PanelButton,
+    FileInput: PanelFileInput,
+    TabPanel: PanelTab,
+    TextInput: PanelTextInput,
+  },
+  LinkToolbar: {
+    Root: Toolbar,
+    Button: ToolbarButton,
+  },
+  Generic: {
+    Form: {
+      Root: (props) => <div>{props.children}</div>,
+      TextInput: TextInput,
+    },
+    Menu: {
+      Root: Menu,
+      Trigger: MenuTrigger,
+      Dropdown: MenuDropdown,
+      Divider: MenuDivider,
+      Label: MenuLabel,
+      Item: MenuItem,
+    },
+    Popover: {
+      Root: Popover,
+      Trigger: PopoverTrigger,
+      Content: PopoverContent,
+    },
+  },
+});
+
+export type ShadCNComponents = {
+  Button: typeof ShadCNButton.Button;
+  DropdownMenu: typeof ShadCNDropdownMenu.DropdownMenu;
+  DropdownMenuCheckboxItem: typeof ShadCNDropdownMenu.DropdownMenuCheckboxItem;
+  DropdownMenuContent: typeof ShadCNDropdownMenu.DropdownMenuContent;
+  DropdownMenuItem: typeof ShadCNDropdownMenu.DropdownMenuItem;
+  DropdownMenuLabel: typeof ShadCNDropdownMenu.DropdownMenuLabel;
+  DropdownMenuSeparator: typeof ShadCNDropdownMenu.DropdownMenuSeparator;
+  DropdownMenuSub: typeof ShadCNDropdownMenu.DropdownMenuSub;
+  DropdownMenuSubContent: typeof ShadCNDropdownMenu.DropdownMenuSubContent;
+  DropdownMenuSubTrigger: typeof ShadCNDropdownMenu.DropdownMenuSubTrigger;
+  DropdownMenuTrigger: typeof ShadCNDropdownMenu.DropdownMenuTrigger;
+  Input: typeof ShadCNInput.Input;
+  Label: typeof ShadCNLabel.Label;
+  Popover: typeof ShadCNPopover.Popover;
+  PopoverContent: typeof ShadCNPopover.PopoverContent;
+  PopoverTrigger: typeof ShadCNPopover.PopoverTrigger;
+  Select: typeof ShadCNSelect.Select;
+  SelectContent: typeof ShadCNSelect.SelectContent;
+  SelectItem: typeof ShadCNSelect.SelectItem;
+  SelectTrigger: typeof ShadCNSelect.SelectTrigger;
+  SelectValue: typeof ShadCNSelect.SelectValue;
+  Tabs: typeof ShadCNTabs.Tabs;
+  TabsContent: typeof ShadCNTabs.TabsContent;
+  TabsList: typeof ShadCNTabs.TabsList;
+  TabsTrigger: typeof ShadCNTabs.TabsTrigger;
+  Toggle: typeof ShadCNToggle.Toggle;
+  Tooltip: typeof ShadCNTooltip.Tooltip;
+  TooltipContent: typeof ShadCNTooltip.TooltipContent;
+  TooltipProvider: typeof ShadCNTooltip.TooltipProvider;
+  TooltipTrigger: typeof ShadCNTooltip.TooltipTrigger;
 };
 
+export const createShadCNComponentContext = (
```

Comment:
this seems a but cumbersome (all the prop passing), isn't this easier with a separate context?